### PR TITLE
Add CheckIngressRule to check-gke-ingress

### PR DESCRIPTION
**Pull request by songrx1997**
/assign @swetharepakula

### CODE REVIEW

File: cmd/check-gke-ingress/app/ingress/rule.go [hunk 1]
```diff
@@ -42,6 +44,8 @@ func CheckServiceExistence(namespace, name string, client clientset.Interface) (
 	return svc, report.Passed, fmt.Sprintf("Service %s/%s found", namespace, name)
 }
 
+// CheckBackendConfigAnnotation checks whether the BackendConfig annotation of
```
Comment (swetharepakula) [inline cmd/check-gke-ingress/app/ingress/rule.go:47]: nit: state what the checks are in the comments

File: cmd/check-gke-ingress/app/ingress/rule.go [hunk 2]
```diff
@@ -31,6 +32,7 @@ import (
 	beconfigclient "k8s.io/ingress-gce/pkg/backendconfig/client/clientset/versioned"
 )
 
+// CheckServiceExistence checks wether a service exists.
```
Comment (swetharepakula) [inline cmd/check-gke-ingress/app/ingress/rule.go:35]: nit: typo whether

File: cmd/check-gke-ingress/app/ingress/rule.go [hunk 3]
```diff
@@ -81,6 +86,15 @@ func CheckHealthCheckTimeout(beConfig *beconfigv1.BackendConfig, svcName string)
...
 
+// CheckIngressRule checks wether an ingress rule has the http field.
+func CheckIngressRule(ingressRule *networkingv1.IngressRule) (*networkingv1.HTTPIngressRuleValue, string, string) {
+	if ingressRule.HTTP == nil {
+		return nil, report.Failed, "IngressRule has no field `http`"
+	}
+	return ingressRule.HTTP, report.Passed, "IngressRule has field `http`"
+}
+
+// getBackendConfigAnnotation gets the BackendConfig annotation from a service.
```
Comment (swetharepakula) [inline cmd/check-gke-ingress/app/ingress/rule.go:97]: Isn't an Ingress with just the default backend defined valid?

File: cmd/check-gke-ingress/app/ingress/rule.go (hunk 1 above)
Comment (songrx1997) [reply cmd/check-gke-ingress/app/ingress/rule.go]: Updated, thanks!

File: cmd/check-gke-ingress/app/ingress/rule.go (hunk 2 above)
Comment (songrx1997) [reply cmd/check-gke-ingress/app/ingress/rule.go]: Updated, thanks!

File: cmd/check-gke-ingress/app/ingress/rule.go (hunk 3 above)
Comment (songrx1997) [reply cmd/check-gke-ingress/app/ingress/rule.go]: This check is for the [IngressRule](https://pkg.go.dev/k8s.io/api@v0.26.2/networking/v1#IngressRule)
It has to have a HTTP ingress rule specified.

**Review by swetharepakula — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**